packages/apex-node/src/utils/types.ts

```diff
@@ -30,3 +38,32 @@ export type ApexDiagnostic = {
 
 export type QueryResult<T = QueryRecord> = { records: T[] };
 export type QueryRecord = { Id: string };
+export type QueryRecords = {
+  totalSize: number;
+  records: IdRecord[];
+};
+
+export type DebugLevelRecord = {
+  ApexCode: string;
+  VisualForce: string;
+};
+
+export type TraceFlagRecord = {
+  Id: string;
+  LogType: string;
+  DebugLevelId: string;
+  StartDate: Date | undefined;
+  ExpirationDate: Date | undefined;
+  DebugLevel: DebugLevelRecord;
+};
+
+export type DataRecordResult = {
+  id?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
could we remove this disable? is there a way for us to determine the error type by forcing an error and walking through it?

Reply:
Over in SDR we specify it as string[] for the callback from tooling.create and tooling.update: https://github.dev/forcedotcom/source-deploy-retrieve/blob/aa79e2642a7aa35b05cb71c9f4cc0b0c21c21123/src/client/toolingApi.ts#L122 

It would be great to test that assumption though, if we can. Let me check with @rcoringrato-sfdc and see if he has any ideas on this one.

Reply:
Roy recommended taking a look at the jsforce docs, which also call out a string array (https://jsforce.github.io/jsforce/doc/global.html#RecordResult). Since SDR also uses string[], I'll add this here for now. Would you be ok if I make a note to test this with the tail command changes, since that would be an easier way to see it in action?

Reply:
perfecto, sounds good to me!